packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -52,9 +52,10 @@ export class TableLayout<T> extends ListLayout<T> {
     this.wasLoading = this.isLoading;
     this.isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
 
-    this.buildColumnWidths();
     let header = this.buildHeader();
     let body = this.buildBody(0);
+    this.stickyColumnIndices = this.collection.columns.filter(c => c.props.isSelectionCell || this.collection.rowHeaderColumnKeys.has(c.key)).map(c => c.index);
```

Comment:
This used to happen with the width calculation but we think it still belongs here and it doesn't have to happen with the width calculation at all, I think it was previously just using the convenient loop that the width calculation provided.